File: ios/IsekaiCameraClient/App/Auth0Config.swift
import Foundation

/// Where the app logs in, and what it asks a token for.
///
/// None of this is secret. A native OAuth client has no client secret — that is
/// the reason it uses PKCE — and the client id travels in the authorize URL of
/// every login, so it is a public identifier like the domain beside it.
///
/// `issuer` and `audience` have to match the Identity API's `AUTH0_ISSUER` and
/// `AUTH0_AUDIENCE`, or the token it receives is rejected.
struct Auth0Config: Equatable {
    var domain = "seera-networks.jp.auth0.com"
    var clientID = "FeDSXYhJsfV1d9v6JyBte874R6En4tok"
    var audience = "https://masque.seera-networks.com/"

    /// `offline_access` is what makes Auth0 return a refresh token, so a session
    /// outlives the access token's few hours. It requires "Allow Offline Access"
    /// on the API in the Auth0 dashboard; without it the login still works and
    /// the app simply asks again when the token expires.
    var scope = "openid profile email offline_access"

    /// Auth0's documented callback shape for a native app. The scheme is the
    /// bundle identifier, which the app registers in its Info.plist, so no other
    /// app on the device can claim the redirect.
    var redirectURI: String {
        "\(Self.bundleID)://\(domain)/ios/\(Self.bundleID)/callback"
    }

    /// The scheme half of `redirectURI`, which is all
    /// `ASWebAuthenticationSession` matches on.
    var callbackScheme: String { Self.bundleID }

    var isConfigured: Bool {
        !domain.isEmpty && !clientID.isEmpty && !audience.isEmpty
    }

    private static let bundleID = Bundle.main.bundleIdentifier ?? "tools.isekai.viewer"
}

File: android-portal/app/src/main/kotlin/tools/isekai/portalclient/Auth0Config.kt
package tools.isekai.portalclient

/**
 * Where the app logs in, and what it asks a token for.
 *
 * Copied from `cameraclient/Auth0Config.kt` -- `domain`/`clientId`/`audience`
 * are shared across every ISEKAI-link app (native OAuth client, no secret, so
 * this is an identifier not a credential).
 *
 * `CALLBACK_SCHEME` is also shared, deliberately, as of Kozuka's "unified
 * callback" decision (2026-08-24): every ISEKAI-link app redirects to the
 * same `isekaiviewer://callback`, rather than each app registering its own
 * entry in Auth0's Allowed Callback URLs list. This app used to have its own
 * `isekaiportal://callback` specifically to avoid Android's disambiguation
 * chooser when two apps claim the same scheme+host -- that tradeoff is now
 * accepted deliberately rather than avoided, in exchange for not needing a
 * dashboard change per app. If both this app and a camera app are installed
 * on the same phone, expect the OS to prompt which app should handle the
 * redirect the first time (or to need re-selecting if the phone's default
 * changes) -- reasonable to route around only if this happens in practice
 * (a click-to-continue or "on this device" default preference does it).
 *
 * **This is a UX question, not a security one, because the flow is
 * Authorization Code + PKCE** (`Auth0Client.kt`): the code this redirect
 * carries is worthless without the verifier, which never leaves this app's
 * memory. A second app that claims `isekaiviewer://callback` -- by the
 * chooser prompt above, or by intercepting before the user picks -- gets a
 * code it cannot redeem, not a stolen session. Worth keeping in mind before
 * "fixing" the chooser prompt by giving this app its own scheme back: that
 * would undo the dashboard-change tradeoff above for a problem PKCE already
 * closed.
 */
object Auth0Config {
    const val DOMAIN = "seera-networks.jp.auth0.com"
    const val CLIENT_ID = "FeDSXYhJsfV1d9v6JyBte874R6En4tok"
    const val AUDIENCE = "https://masque.seera-networks.com/"

    /** `offline_access` is what makes Auth0 return a refresh token. */
    const val SCOPE = "openid profile email offline_access"

    /**
     * The scheme half of the redirect URI. Must also match the
     * `<data android:scheme=...>` in AndroidManifest.xml's intent-filter for
     * `.MainActivity`. Shared with every other ISEKAI-link app -- see the
     * class doc comment above.
     */
    const val CALLBACK_SCHEME = "isekaiviewer"
    const val REDIRECT_URI = "$CALLBACK_SCHEME://callback"
}
